모임 단건 조회 API 생성

hyeonnkang opened the pull request:
## 작업 사항
close #45 
- 모임 단건 조회 API 구현

## 참고 자료

- 참고자료 (링크 등)

## 체크리스트

- [x]  PR의 제목은 팀 내 규칙을 준수하여 알맞게 작성하였는가?
- [x]  Merge 하는 브랜치가 올바른가? (`main` branch에 실수로 PR 생성 금지)
- [x]  팀의 코딩 컨벤션을 준수하는가?
- [ ]  PR과 관련없는 변경사항이 들어가지는 않았는가?
- [x]  내 코드에 대한 자기 검토가 되었는가?
- [x]  Reviewers, Assignees, Lables, Project, Milestone은 적절하게 선택하였는가?
- [x]  관련한 issue를 닫아야 하는지 점검해보고 적용했는가?

hyeonnkang committed:
Merge remote-tracking branch 'origin/develop' into feature/#45-meet-findById

hyeonnkang committed:
refactor: #45 테스트 코드 내의 함수명 변경

MeetService 내의 createMeet 메소드와 이름이 같은 함수가 테스트 클래스 내부에 있었다.
추후 두 메소드를 헷갈릴 여부가 있을 것 같아 함수명을 변경하기로 했다.
테스트 클래스의 createMeet, createMember 메소드의 이름을 getMeet, getMember로 변경했다.

hyeonnkang committed:
feat: #45 'Meet' 모임 단건 조회 Service 구현 및 테스트

Service Layer에서 모임 단건 조회 기능을 구현 및 테스트 완료 했습니다.

hyeonnkang committed:
feat: #45 모임 단건 조회 API 생성 및 테스트

- 모임 조회 시 모임의 정보를 담아 반환하는 MeetInfoGetResponse를 생성해주었다.
- 기존 MemberDto와 MemberInfoGetresponse 내의 Address 필드를 AddressDto 필드로 교체해주었다.
- 추가로 MeetService에서 createMeet 메소드에 대한 주석이 잘못 작성되어있어서 이 부분을 변경했다.

Wo-ogie commented on src/main/java/com/umc/withme/dto/member/MemberDto.java:
```diff
@@ -24,7 +25,7 @@ public class MemberDto {
     private TotalPoint totalPoint;
 
     public static MemberDto of(Long id, Address address, String email, String phoneNumber, String nickname, LocalDate birth, Gender gender, TotalPoint totalPoint) {
-        return new MemberDto(id, address, email, phoneNumber, nickname, birth, gender, totalPoint);
+        return new MemberDto(id, AddressDto.from(address), email, phoneNumber, nickname, birth, gender, totalPoint);
     }
```
일반적으로 dto의 `of()` method의 parameter로는 field를 그대로 받아주시면 됩니다. `Address`는 entity로 받아도 상관 없을 것 같긴 한데 혹시 번거롭지 않으시면 dto로 바꿔주실 수 있을까요?
ccbc43cd 의 `MemberDto` 코드를 참고해주시면 좋을 것 같습니다! 😊

Wo-ogie commented on src/main/java/com/umc/withme/repository/MeetAddressRepository.java:
```diff
@@ -6,4 +6,5 @@
 import java.util.List;
 
 public interface MeetAddressRepository extends JpaRepository<MeetAddress, Long> {
+    List<MeetAddress> findByMeet_Id(Long meetId);
```
`findAllByMeet_Id`로 하여 리스트를 조회하는 method라는 것을 강조해주시면 좋을 것 같습니다.

Wo-ogie commented on src/main/java/com/umc/withme/service/MeetService.java:
```diff
@@ -50,6 +52,24 @@ private Address getAddress(AddressDto dto) {
+     * @return
+     */
+    public MeetDto findById(Long meetId) {
+        Meet meet = meetRepository.findById(meetId)
+                .orElseThrow(EntityNotFoundException::new);
```
`createMeet()`와 `findById()`에서 다음 코드가 중복되고 있네요.
```java
meetRepository.findById(meetId).orElseThrow(EntityNotFoundException::new);
```

이럴 때는 중복되는 코드를 다음과 같이 method로 추출하여 중복을 줄여보는 것도 좋을 것 같습니다. 이러면 throw 해야 할 exception이 변경 될 때 해당 기능을 사용하는 모든 method에 전체 적용이 되는 장점이 있을 것 같아요.
```java
private Meet getMeetById(Long meetId) {
    return meetRepository.findById(meetId).orElseThrow(EntityNotFoundException::new);
}
```

또한 `EntityNotFoundException`은 `NicknameNotFoundException`, `EmailNotFoundException`처럼 `MemberIdNotFoundException`이라는 custom exception을 만들어서 던져주는 것이 좋을 것 같아요.

hyeonnkang replied:
1. `createMeet()` 내에서 `memberRepository.findById` 와 `findById()` 내의 `meetRepository.findById`를 중복으로 보신건지 궁금합니다! 

2. custom exception은 `MeetIdNotFoundException`으로 만들었습니다! `MemberIdNotFoundException`은 인증기능 구현되면 따로 처리된다는 TODO 메세지가 있어서 따로 만들지 않았어요.

Wo-ogie replied:
앗 1번은 제가 잘못 본 게 맞네요 ㅠㅠ 죄송합니다. 안하셔도 돼용
2번은 그러면 `MeetNotFoundException`으로 해주시면 될 것 같습니다. 
Member랑 Meet랑 잘못 봤네요 😅

hyeonnkang replied:
감사합니다! 위에 말씀해주신 부분 전부 반영해서 push 했습니다!

hyeonnkang committed:
feat: #45 코드리뷰 반영

- MemberDto이 of 메소드에서 AddressDto 필드 값을 파라미터로 그대로 받도록 변경
- MeetAddressRepository에서 결과가 리스트임을 명시적으로 알 수 있도록 All 네이밍 추가
- MeetIdNotFoundException 생성

Wo-ogie commented:
Custom exception 만드셨으면 `ExceptionType` enum class에도 해당 내용 반영해주셔야 합니다! 😊

hyeonnkang committed:
feat: #45 MEET_ID_NOT_FOUND_EXCEPTION 추가

생성한 custom class를 ExceptionType 에 MEET_ID_NOT_FOUND_EXCEPTION으로 반영해주었다.

hyeonnkang commented:
> Custom exception 만드셨으면 `ExceptionType` enum class에도 해당 내용 반영해주셔야 합니다! 😊

앗 넵 확인하고 반영했습니다!

hyeonnkang committed:
feat: #45 MEET_ID_NOT_FOUND_EXCEPTION 에러코드 변경

Meet 에러 코드를 2000번대로 하고 100의 자리수는 Http Status 코드로 하도록 2400 번 에러코드로 수정했다.

Wo-ogie commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -41,4 +43,23 @@ public ResponseEntity<DataResponse<MeetCreateResponse>> createMeet(
+     * @param meetId
+     * @return 조회한 모임의 정보를 담은 MeetInfoGetResponse를 data에 담아서 반환한다.
+     */
+    @GetMapping("/meet")
+    public ResponseEntity<DataResponse<MeetInfoGetResponse>> getMeet(@RequestParam Long meetId){
```
```suggestion
    @GetMapping("/meet/{meetId}")
    public ResponseEntity<DataResponse<MeetInfoGetResponse>> getMeet(@PathVariable Long meetId){
```
이 API는 query string보다는 path variable이 더 낫지 않을까요?

hyeonnkang replied:
생각해보니 그게 더 적합할 것 같네요. 해당 사항 반영했습니다!

hyeonnkang committed:
feat: #45 모임 단건 조회 API meetId 받아오는 방법 변경

모임 단건 조회 시 meetId를 받아오는 방법이 기존에는 RequestParam 이었으나 PathVariable 로 변경했습니다.

Wo-ogie reviewed (approved):
다음부터는 될 수 있으면 method 설명에 `@param`, `@return`도 작성해주시면 좋을 것 같습니다! 수고하셨어요 😊

hyeonnkang committed:
Merge branch 'develop' into feature/#45-meet-findById

hyeonnkang committed:
feat: #45 erd 변경 (Meet 에서 leader_id 삭제)

Meet 테이블에서 leader_id가 create_by와 겹쳐 leader_id를 삭제했다. 이에 따라 Meet과 Member 사이의 One to One 관계 하나가 삭제 되었고 해당 사항이 반영된 erd 이미지를 변경해주었다.

hyeonnkang committed:
feat: #45 'Meet' 인증 적용 및 swagger 주석 추가

변경된 erd 에 따라 모임에서 leader 삭제했다.
해당 유저 네임 미 조회 시 exception 교체 EntityNotFoundException -> UnauthorizedException 로 변경했다.
